Instagram/View Controllers/ComposeViewController.m

```diff
@@ -55,8 +83,39 @@ - (IBAction)cancel:(id)sender {
 }
 
 - (IBAction)share:(id)sender {
-    [Post postUserImage:[self.imageView image] withCaption:[self.captionField text] withCompletion:nil];
-    [self clear];
+    [self startProgress];
+    
+    [Post postUserImage:[self.imageView image] withCaption:[self.captionField text] withCompletion:^(BOOL succeeded, NSError * _Nullable error) {
+        if (succeeded) {
+            [self endProgress];
+        } else {
+            [self clear];
```

Comment:
This method name could be more descriptive. It's not clear to me at the callsite what it's doing.